### PR TITLE
Support point in time in async search

### PR DESCRIPTION
**Pull request by dnhatn**
This commit integrates point in time into async search and ensures that it works correctly with security enabled.

Relates #61062

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/search/action/RestOpenPointInTimeAction.java [hunk 1]
```diff
@@ -28,7 +28,9 @@ public String getName() {
 
     @Override
     public List<Route> routes() {
-        return List.of(new Route(POST, "/{index}/_pit"));
+        return List.of(
+            new Route(POST, "/{index}/_pit"),
+            new Route(POST, "/_pit"));
```
Comment (jimczi) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/search/action/RestOpenPointInTimeAction.java:33]: Maybe we should disable creating a `_pit` for all indices ?

**Review by jimczi — APPROVED** [review]
It looks great, thanks @dnhatn !

**Comment by dnhatn** [comment]
> Maybe we should disable creating a _pit for all indices ?

@jimczi Thank you for your review. I am merging this PR to unblock the work in ccs. I will discuss it with you tomorrow.

**Review by albertzaharovits — COMMENTED** [review]
LGTM

**Pull request merged**